tap_github/repository_streams.py

```diff
@@ -1907,6 +1907,457 @@ def query(self) -> str:
     ).to_dict()
 
 
+class DiscussionsStream(GitHubGraphqlStream):
+    """Defines stream fetching discussions from each repository."""
+
+    name = "discussions"
+    query_jsonpath = "$.data.repository.discussions.nodes.[*]"
+    primary_keys: ClassVar[list[str]] = ["id"]
+    replication_key = "updated_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys: ClassVar[list[str]] = ["repo", "org"]
+    ignore_parent_replication_key = False
+    use_fake_since_parameter = True
+
+    def get_url_params(
+        self,
+        context: dict | None,
+        next_page_token: Any | None,  # noqa: ANN401
+    ) -> dict[str, Any]:
+        """Return a dictionary of values to be used in URL parameterization."""
+        params = super().get_url_params(context, next_page_token)
+        self.logger.info(f"URL Params: {params}")
+        self.logger.info(f"Context: {context}")
+        self.logger.info(f"State: {self.get_starting_replication_key_value(context)}")
+        return params
+
+    @property
+    def query(self) -> str:
+        """Return dynamic GraphQL query."""
+        # Graphql id is equivalent to REST node_id. To keep the tap consistent, we rename "id" to "node_id".  # noqa: E501
```

Comment:
May as well add this to the docstring:
```suggestion
        """
        Return dynamic GraphQL query.
        Note: To keep the tap consistent, we rename id to node_id and databaseId to id.
        """
```
or however the linter wants to style it.
(and same for the other streams)